scripts/sign_dlls.py

```diff
@@ -12,107 +12,76 @@
 
 
 from pathlib import Path
-from typing import Generator
+from typing import List
 import shutil
 import argparse
 import json
 
 
-def determine_to_sign_dll_names(build_dir: Path) -> Generator[str, None, None]:
+def get_paths_of_dlls_to_sign(build_dir: Path) -> List[Path]:
     """
     Gather all the dll names which need to be signed.
 
-    We assume that the only dlls produced by the solution are the
-    ones created by their corresponding .vcxproj and .csproj.
+    It is assumed that the only dlls produced by the solution are the ones created
+    by their corresponding .vcxproj and .csproj projects in Release mode.
 
     Args:
         build_dir (Path): Path to the build dir
 
     Returns:
-        Generator[str, None, None]: Generator containing the dll names to sign.
+        List[Path]: List of dll paths to sign.
     """
-    vs_proj_types = ("*.vcxproj", "*.csproj")
-    vs_projs = []
+    vs_proj_types = ("vcxproj", "csproj")
+    vs_proj_paths = []
     for vs_proj_type in vs_proj_types:
-        vs_projs.extend((build_dir / Path("libs")).glob(f"**/{vs_proj_type}"))
-    return (
-        x.with_suffix(".dll").name
-        for x in vs_projs
-        if "test" not in x.with_suffix(".dll").name.lower()
-    )
-
-
-def get_corresponding_path(build_dir: Path, dll_name: str) -> Path:
-    """
-    Get the dll path corresponding with the dll_name from the provided build_dir.
-
-    Args:
-        build_dir (Path): Path to the build dir
-        dll_name (str): Name of the dll
-    """
-    # this hack is required to keep the capitalisation
-    return next(
-        (build_dir / Path("libs")).glob("**/Release/{}".format(dll_name))
-    ).parent / Path(dll_name)
+        vs_proj_paths.extend(Path(f"{build_dir}/libs").glob(f"*/*.{vs_proj_type}"))
+        vs_proj_paths.extend(Path(f"{build_dir}/libs").glob(f"*/*/*.{vs_proj_type}"))
 
+    dll_paths = []
+    for vs_proj_path in vs_proj_paths:
+        parent = vs_proj_path.parent
+        stem = vs_proj_path.stem
+        dll_path = Path(f"{parent}/Release/{stem}.dll")
+        if dll_path.exists() and "test" not in stem.lower():
+            dll_paths.append(dll_path)
```

Comment:
extend for consistency?

Reply:
List of paths if not iterable.
```
dll_paths.extend(dll_path)
TypeError: 'WindowsPath' object is not iterable
```
Extend is used earlier to add multiple elements at once to an iterable list.